pkgs/development/compilers/gcc/common/strip-attributes.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv }:
+
+{
+  # Note [Cross-compiler stripping]
+  # gcc requires delicate stripping as it installs ELF files for both
+  # HOST and TARGET platforms. It requires according strip tool otherwise
+  # strip could remove sections it's not aware of.
+  # Example ARM breakage by x86_64 strip: https://bugs.gentoo.org/697428
+  #
+  # Let's recap the file layout for directories with object files for a
+  # cross-compiler (host != target):
+  # `- bin: HOST
+  #    lib/*.{a,o}: HOST
+  #      `- gcc/<TARGET>/<VERSION>/*.{a,o}: TARGET
+  #                               `- plugin/: HOST
+  #  `- lib{,32,64,x32}: symlink to lib or identical layout
+  #  `- libexec/: HOST
+  #  `- <TARGET>/: TARGET
+  #
+  # (host == target) has identical directory layout.
+
+  # The rest of stripDebugList{Host,Target} will be populated in
+  # postInstall.
+  stripDebugList = [ "bin" "libexec" ];
+  stripDebugListTarget = [ stdenv.targetPlatform.config ];
+
+  preFixup = ''
+    # Populate most delicated lib/ part of stripDebugList{,Target}
+    updateDebugListPaths() {
```

Comment:
Did you put this into a function so that it can be re-used later in some other phase? If so I think we should use a hook instead.

Reply:
The function should be used only here. I used function only for `local oldOpts="$(shopt -p nullglob)"` variable scoping.

Reply:
Would a subshell also scope it?

Reply:
Function's main effect is to extend variables used by a `pkgs/build-support/setup-hooks/strip.sh`:
```
      stripDebugList="$stripDebugList ''${hostFiles[*]}"
      stripDebugListTarget="$stripDebugListTarget ''${targetFiles[*]}"
```

Thus subshell would need to be applied selectively on a subset of the function's code.